mgmt-lambda/handlers/updateHandler.ts

```diff
@@ -63,20 +68,20 @@ async function updateCloudFrontConfig(
   const cfConfig: GetDistributionConfigCommandOutput = await cloudFrontClient.send(getConfigCommand)
 
   if (!cfConfig.ETag || !cfConfig.DistributionConfig) {
-    return handleFailure('CloudFront distribution not found')
+    throw new ApiException(ErrorCode.CloudFrontDistributionNotFound)
   }
 
   const cacheBehaviors = cfConfig.DistributionConfig.CacheBehaviors
   const fpCbs = cacheBehaviors?.Items?.filter((it) => it.TargetOriginId === 'fpcdn.io')
   if (!fpCbs || fpCbs?.length === 0) {
-    return handleFailure('Cache behavior not found')
+    throw new ApiException(ErrorCode.CacheBehaviorNotFound)
   }
   const cacheBehavior = fpCbs[0]
   const lambdas = cacheBehavior.LambdaFunctionAssociations?.Items?.filter(
     (it) => it && it.EventType === 'origin-request' && it.LambdaFunctionARN?.includes(`${lambdaFunctionName}:`),
   )
   if (!lambdas || lambdas?.length === 0) {
-    return handleFailure('Lambda function association not found')
+    throw new ApiException(ErrorCode.LambdaFunctionAssociationNotFound)
   }
   const lambda = lambdas[0]
   lambda.LambdaFunctionARN = latestFunctionArn
```

Comment:
Let's make this more explicit. The update action is not obvious, the code `DistributionConfig: cfConfig.DistributionConfig,` depends on the `lambda` variable but that is not easy to follow. We can create a variable called `newDistributionConfig` and use it instead.

Reply:
Maybe we should refactor this in another PR because this requires rewrite the logic of function itself and it's too out of scope for this PR. Skipping for now.